受信したコマンドの処理の振り分けができるようになった。

youkan-man opened the pull request.

yu-san committed:
応答状態がわかるよう型をenumに変更

共通の型をcommon.hに集約

yu-san committed:
コマンド待ち受け処理

yu-san committed:
interrupt名を変更したことで動かなくなっていた不具合の対応

yu-san committed:
・コマンド受信

　確認するには、暫定対応としてcomunicate.cの#define STX_DEF (0x02)→(0x30)に変更する

・受信コマンドに応じた出力対応
（本番対応では、無線通信機のヘッダー情報で正しく判定できないため、コマンドに応じた出力は未試験）

youkan-man commented on DreamSat/Camera.cydsn/common.h:
```diff
@@ -0,0 +1,29 @@
+#define DREAMSAT_CAMERA_COMMON_H_
+
+typedef unsigned char cmd_digit;
+
```
unsigned char ⇒ uint8

y-okumura replied:
https://ttsuki.github.io/styleguide/cppguide.ja.html#Type_Names

> 型の名前は大文字で始まり、単語の区切りごとに大文字にします。アンダースコアは使いません。たとえば、MyExcitingClass、MyExcitingEnumとします。

CommandDigitとか。

youkan-man commented on DreamSat/Camera.cydsn/common.h:
```diff
@@ -0,0 +1,29 @@
+    ret_success,
+    ret_error
+}rettype;
+
```
retやreturnの名前は言葉として意味をもっていない。
rettype ⇒ StatusCode/Statusなどに変更する。
型名は大文字から初めて、方の区切りごとに大文字にする。

y-okumura replied:
```
kStatusCodeSuccess,
kStatusCodeError
```
です

y-okumura replied:
https://ttsuki.github.io/styleguide/cppguide.ja.html#Enumerator_Names

あと、StatusCodeは無くて良さそう。

```
kSuccess,
kError
```
ですね。

y-okumura replied:
kすごい邪魔ですね。
統一取れたら全部大文字のがいい気がしてきました…

youkan-man commented on DreamSat/Camera.cydsn/common.h:
```diff
@@ -0,0 +1,29 @@
+    cmd_panic,
+    cmd_parroting
+}cmdtype;
+
```
型名cmdtypeは型名の命名規則にしたがって定義する。
cmdtype ⇒ CmdType

y-okumura commented on DreamSat/Camera.cydsn/common.h:
```diff
@@ -0,0 +1,29 @@
+    cmd_parroting
+}cmdtype;
+
+void InitializeCommUart();
```
https://ttsuki.github.io/styleguide/cppguide.ja.html#Function_Comments

> ほぼすべての関数宣言において、その宣言の直前に、関数の目的と使用方法に関するコメントをつけます。

y-okumura commented on DreamSat/Camera.cydsn/comunicate.c:
```diff
@@ -10,12 +10,51 @@
 #include "comunicate.h"
 #include "project.h"
 
+CY_ISR_PROTO(Comm_Rx_Intr);
```
https://ttsuki.github.io/styleguide/cppguide.ja.html#Function_Names

> 通常、関数名は大文字で始まり、単語の境目に大文字を使用します(「Camel Case」や「Pascal Case」とも呼ばれています)。 これらの名前にはアンダースコアは使いません。

y-okumura replied:
https://ttsuki.github.io/styleguide/cppguide.ja.html#General_Naming_Rules

> 記述的な名前をつけてください。省略形は避けてください。
> 一般に広く知られている省略形を使うことはOKであることに注意してください

Commは微妙、RxはOK（というか正式名称のほうが知りません）、IntrはNGかなぁ、と思います

y-okumura commented on DreamSat/Camera.cydsn/comunicate.c:
```diff
@@ -10,12 +10,51 @@
 #include "comunicate.h"
 #include "project.h"
 
+CY_ISR_PROTO(Comm_Rx_Intr);
```
https://ttsuki.github.io/styleguide/cppguide.ja.html#Forward_Declarations

> 前方宣言は可能な限り避けましょう。 必要ならば単にヘッダを#includeしてください。

関数の実体箇所で CY_ISR するだけでよいかと。

youkan-man commented on DreamSat/Camera.cydsn/common.h:
```diff
@@ -0,0 +1,29 @@
+{
+    ret_success,
+    ret_error
+}rettype;
```
列挙体の命名規則は定数名と合わせる。
ret_success ⇒ kRetSuccess

youkan-man commented on DreamSat/Camera.cydsn/common.h:
```diff
@@ -0,0 +1,29 @@
+typedef unsigned char cmd_digit;
+
+typedef enum rettypetag
+{
```
タグ名は省略する。
rettypetag ⇒ 削除

y-okumura reviewed (changes requested):
命名規約を中心に書いてみました

youkan-man committed:
Designerで設定できる名前を変更し、それにあわせてプログラムを修正

youkan-man committed:
UARTデバイス名の変更にともない、それを利用するファイル名の変更

communicate.h
communicate.c
earth.h
earth.c

youkan-man committed:
ビルドwarningを解消

youkan-man committed:
InitializeCommUartの関数定義がCommon.hに記載されていたので移動

youkan-man committed:
ファイル名の変更/依存関係の調整

earth.h/earth.c
をそれぞれ
communicate_to_earth.h
commnnicate_to_earth.c
に変更。

communicate_to_earthがincludeしていたoperation.hの依存関係の調整
operation.h内で定義されている関数を、個々のヘッダファイルに移動。
main.cではoperation.hに加え、個々のオペレーションファイルをincludeするように修正。

youkan-man committed:
communication_to_earthの初期化関数名の変更。

youkan-man committed:
Enumの型定義値を変更

rettype
⇒StatusCode
cmdtype
⇒CommandType

youkan-man committed:
communicate_to_earth.hの２インクルード防止の定義名を修正

y-okumura commented on DreamSat/Camera.cydsn/convert_cmd.c:
```diff
@@ -10,12 +10,18 @@
 unsigned short ConvertCommand(cmd_digit* command)
 {
     unsigned short cmd_type = 0;
-    if (command) {
```
commandは8要素のバイト列で、文字列ではない(0終端されない)のでstrcmp は使えないはずです。

youkan-man committed:
cmd_digitを命名規則に従って修正

youkan-man committed:
reply_digitを命名規則に従って修正

youkan-man committed:
戻り値の型をenumの型定義に変更/unsigned charをuint8に変更

youkan-man committed:
コマンド判定箇所の文字列比較関数をメモリ比較関数に変更

youkan-man committed:
関数名の変更

RecieveCommand
SendReply
⇒ RecieveFromEarth
⇒ SendToEarth

youkan-man committed:
CommandDigit/ReplyDigitの型定義を共通の型定義Byteに変更

youkan-man committed:
ヘッダ上でインクルードしていたproject.hを削除

youkan-man committed:
インタラプト関数内でデータ受信時のパラメータチェックを追加。

youkan-man committed:
コマンド比較対象として宣言している変数をconst型に変更

youkan-man committed:
インタラプト関数内でのパラメータチェックの不具合繻修正

youkan-man committed:
CheckDigitのコメント追加。

youkan-man committed:
メインループ追加

youkan-man committed:
地球からの受信処理で受信バッファの初期化を追加

youkan-man committed:
地球への送信バッファサイズの指定を、定数からDefine値に変更

youkan-man committed:
Revert "メインループ追加"

This reverts commit 8545986c3596196645415a078a635015c745b973.

youkan-man committed:
Revert "地球からの受信処理で受信バッファの初期化を追加"

This reverts commit c6c8510e2d3de9f4ff0c41fbf794de31599e85ed.

youkan-man committed:
memcmpの戻り値が0のときに条件が真になるように修正
